single-kernel/nbody.cpp

```diff
@@ -105,41 +106,19 @@ class NBody
       host_resulting_velocities[i] = new_v;
     }
 
-    double deviation = std::sqrt(
-        calculateSquaredDifference(host_resulting_particles.data(), resulting_particles.get_pointer(), particles.size()) +
-        calculateSquaredDifference(host_resulting_velocities.data(), resulting_velocities.get_pointer(), particles.size()));
-
-    return deviation < 1.e-6;
+    constexpr float_type maxErr = 10.f * std::numeric_limits<float_type>::epsilon();
+    return checkResults(
+               host_resulting_particles.begin(), host_resulting_particles.end(), resulting_particles.begin(), maxErr) &&
```

Comment:
hipSYCL does not yet have the `accessor::begin()/end()` API, so this PR breaks hipSYCL support. Can you perhaps just use pointers instead?

Reply:
Of course. Just changed that. Thanks for the review, Aksel.